docs/gen_ref_pages.py

```diff
@@ -59,40 +84,124 @@ def insert_blocks_catalog(generated_file):
     for module_nesting, block_names in module_blocks.items():
         module_path = " ".join(module_nesting)
         module_title = module_path.replace("_", " ").title()
-        generated_file.write(f"### {module_title} Module\n")
+        generated_file.write(
+            f"## [{module_title} Module][{COLLECTION_SLUG}.{module_path}]\n"
+        )
         for block_name in block_names:
             generated_file.write(
-                f"- **[{block_name}][{COLLECTION_SLUG}.{module_path}.{block_name}]**\n"
+                f"[{block_name}][{COLLECTION_SLUG}.{module_path}.{block_name}]\n"
             )
-        generated_file.write(
-            dedent(
-                f"""
-                To load the {block_name}:
-                ```python
-                from prefect import flow
-                from {COLLECTION_SLUG}.{module_path} import {block_name}
-
-                @flow
-                def my_flow():
-                    my_block = {block_name}.load("MY_BLOCK_NAME")
-
-                my_flow()
-                ```
-                """
+            generated_file.write(
+                dedent(
+                    f"""
+                    To load the {block_name}:
+                    ```python
+                    from prefect import flow
+                    from {COLLECTION_SLUG}.{module_path} import {block_name}
+
+                    @flow
+                    def my_flow():
+                        my_block = {block_name}.load("MY_BLOCK_NAME")
+
+                    my_flow()
+                    ```
+                    """
+                )
             )
-        )
 
 
-readme_path = Path("README.md")
-docs_index_path = Path("index.md")
+blocks_catalog_path = Path("blocks_catalog.md")
+with mkdocs_gen_files.open(blocks_catalog_path, "w") as generated_file:
+    insert_blocks_catalog(generated_file)
 
-with open(readme_path, "r") as readme:
-    with mkdocs_gen_files.open(docs_index_path, "w") as generated_file:
-        for line in readme:
-            if line.startswith("Visit the full docs [here]("):
-                continue  # prevent linking to itself
-            if line.startswith("## Resources"):
-                insert_blocks_catalog(generated_file)
-            generated_file.write(line)
+# Examples Catalog page
 
-    mkdocs_gen_files.set_edit_path(Path(docs_index_path), readme_path)
+
+def skip_parsing(name, obj, module_nesting):
+    try:
+        wrong_module = not to_qualified_name(obj).startswith(module_nesting)
+    except AttributeError:
+        wrong_module = False
+    return obj.__doc__ is None or name.startswith("_") or wrong_module
+
+
+def skip_code_example(code_example: str) -> bool:
+    """
+    Skips the code example if it's just showing how to load a Block.
+    """
+    return re.search(r'\.load\("BLOCK_NAME"\)\s*$', code_example.rstrip("`"))
+
+
+def get_code_examples(obj: Any) -> Set[str]:
```

Comment:
I think you can type parameters as `Module`?
```suggestion
def get_code_examples(obj: Module) -> Set[str]:
```